src/elements/OcTabbed.vue

```diff
@@ -0,0 +1,138 @@
+<template>
+  <div>
+    <div
+      class="oc-tabbed"
+      role="tablist"
+      ref="tablist"
+      @click="clickHandler"
+      @keydown="keydownHandler"
+    >
+      <slot name="tabs"></slot>
+    </div>
+    <div ref="panellist">
+      <slot name="panels"></slot>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: "oc-tabbed",
+  status: "prototype",
+  release: "1.0.0",
+  data() {
+    return {
+      tabs: [],
+      panels: [],
+      activeTab: `tab-${this.initial}`,
+    }
+  },
+  props: {
+    initial: {
+      type: String,
+      required: true,
+      default: null,
+    },
+  },
+  mounted() {
+    for (const slotEl of this.$slots.tabs) {
+      if (slotEl.componentOptions && slotEl.componentOptions.tag === "oc-tabbed-tab") {
+        if (slotEl.componentInstance.$el.disabled) return
+        this.tabs.push(slotEl.componentInstance)
+      }
+    }
+
+    for (const slotEl of this.$slots.panels) {
+      if (slotEl.componentOptions && slotEl.componentOptions.tag === "oc-tabbed-panel") {
+        this.panels.push(slotEl.componentInstance)
+      }
+    }
+
+    this.tabs.forEach((tab, i) => {
+      tab.$el.setAttribute("aria-controls", this.panels[i].id)
+    })
+
+    this.panels.forEach((panel, i) => {
+      panel.$el.setAttribute("aria-labelledby", this.tabs[i].id)
+    })
+
+    this.changeTab(this.activeTab)
+  },
+  methods: {
+    changeTab(tab) {
+      this.$nextTick(() => {
+        const tabEl = this.$refs.tablist.querySelector(`#${tab}`)
+
+        this.panels.forEach(panel => {
+          panel.$el.hidden = true
+        })
+
+        this.$refs.panellist.querySelector(`[aria-labelledby=${tab}]`).hidden = false
+
+        this.$refs.tablist.querySelectorAll("[aria-selected]").forEach(el => {
+          el.removeAttribute("aria-selected")
+          el.setAttribute("tabindex", "-1")
+        })
+
+        tabEl.setAttribute("aria-selected", "true")
+        tabEl.setAttribute("tabindex", "0")
+        tabEl.focus()
+
+        this.activeTab = tab
+        this.$emit("activeTab", tab.replace("tab-", ""))
+        this.$emit("click")
+      })
+    },
+    clickHandler(e) {
+      if (!e.target.attributes["aria-controls"]) return
+      this.changeTab(e.target.id)
+    },
+    keydownHandler: function(e) {
+      const activeElem = this.$refs.tablist.querySelector(`#${this.activeTab}`)
```

Comment:
using query selector feels do un-vuejs-ish ... this is what vuejs has refs for ..... or am I missing anything?

Reply:
I can't really use a ref here since the name is dynamic, I think

Reply:
indeed, seems not possible. refs are limited to the current component: https://stackoverflow.com/a/42814625